websites/H/HackTheBox/presence.ts

```diff
@@ -0,0 +1,218 @@
+		},
+		{
+			path: "/users/{}",
+			details: "Looking at profile",
+			stateMethod: "getPersonalProfileDetails",
+		},
+		{
+			path: "/tracks",
+			details: "Tracks",
+			state: "Browsing tracks...",
+		},
+	],
+	presence = new Presence({
+		clientId: "1042105891681472595",
+	});
+
+function getHomePageDetails() {
+	return `${
+		document
+			.querySelectorAll(".text-left.font-size15.col")[0]
+			.querySelectorAll(".color-green")[0].textContent
+	}, Rank: ${
+		document.querySelectorAll(".htb-text2.font-size15.text-left.mb-0")[0]
+			.textContent
+	}`;
+}
+
+function getPersonalProfileDetails() {
+	return document.querySelectorAll(".htb-subtitle.greenOnHover")[0].textContent;
+}
+
+function getMachineDetails() {
+	const diff = document.querySelectorAll(
+			".d-inline-block.pl-2.pr-2.borderRadius13.label-reset.no-space.color-main.htb-label.undefined.letterSpacingundefined.borderRadius4"
+		)[0].textContent,
+		name = document
+			.querySelectorAll(
+				".d-inline-block.pl-2.pr-2.borderRadius13.label-reset.no-space.color-white.fontSemiBold.letterSpacing0.borderRadius4"
+			)[1]
+			.querySelectorAll("span")[0].textContent,
+		status = document
+			.querySelectorAll(".htb-label2.offline-text.text-left.pl-3")[0]
+			.textContent.includes("offline")
+			? "offline"
+			: "online";
+
+	return `${name} (${diff}) - ${status}`;
+}
+
+function getChallengeDetails() {
+	const name = document
+		.querySelectorAll(
+			".d-inline-block.pl-2.pr-2.borderRadius13.label-reset.no-space.color-white.fontSemiBold.letterSpacing0.borderRadius4"
+		)[1]
+		.querySelectorAll("span")[0].textContent;
+
+	return `${name} - ${
+		document.querySelectorAll(".htb-label2.offline-text.text-left.pl-3")[0]
+			.textContent
+	}`;
+}
+
+function getUserId() {
+	return document.cookie
+		.split(";")
+		.find(item => item.includes("ajs_user_id"))
+		.split("=")[1];
+}
+
+function findPresence() {
+	const item = presences.find(item => {
+		if (item.path.includes("{}")) {
+			if (
+				new RegExp(item.path.replace("{}", ".+")).test(window.location.pathname)
+			)
+				return item;
+		} else if (document.location.pathname.endsWith(item.path)) return item;
+		else if (document.location.pathname.includes(item.path)) return item;
+		else return null;
+	});
+
+	return item;
+}
+
+interface IMethods {
+	[key: string]: () => string;
+}
+
+const methods: IMethods = {
+	getHomePageDetails,
+	getMachineDetails,
+	getChallengeDetails,
+	getPersonalProfileDetails,
+};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {};
+
+	for (let i = 0; i < presences.length; i++) {
+		const item = findPresence();
+
+		if (item) {
+			presenceData.details = item.details;
+			if (item.stateMethod)
+				presenceData.state = methods[item.stateMethod as keyof IMethods]();
+			else presenceData.state = item.state;
+		}
+	}
+
+	presenceData.largeImageKey = "icon";
+	presenceData.buttons = [
+		{
+			label: "View Page",
+			url: document.location.href,
+		},
```

Comment:
This has the possibility of adding a button that leads to the home page, which is [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets). Adding a button to every single page is not good practice for this repo.